File: app/schemas/task.py
from pydantic import BaseModel, ConfigDict
from datetime import datetime

class TaskCreate(BaseModel):
    task_name: str
    goal_id: int

class TaskRead(BaseModel):
    task_id: int
    task_name: str
    created_at: datetime
    goal_id: int

    model_config = ConfigDict(from_attributes=True)

File: web-app/src/types/task.ts
export interface TaskCreate {
  task_name: string;
  goal_id: number;
}

export interface TaskUpdate {
  task_name?: string | null;
}

export interface TaskRead {
  task_id: number;
  task_name: string;
  created_at: string;
  goal_id: number;
}
